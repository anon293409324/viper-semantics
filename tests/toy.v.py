def foo(y:num):
    z = y
