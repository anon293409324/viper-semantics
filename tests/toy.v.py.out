[seq,
  [return,
    0,
    [lll,
      [seq,
        [seq,
          [mstore, 28, [calldataload, 0]],
          [mstore, 32, 1461501637330902918203684832716283019655932542976],
          [mstore, 64, 170141183460469231731687303715884105727],
          [mstore, 96, -170141183460469231731687303715884105728],
          [mstore, 128, 1701411834604692317316873037158841057270000000000],
          [mstore, 160, -1701411834604692317316873037158841057280000000000]],
        # Line 1
        [if,
          [eq, [mload, 0], 3650092561 <foo>],
          [seq,
            [calldatacopy, 320, 4, 32],
            [assert, [iszero, callvalue]],
            /* checking num input */ 
            [clamp,
              [mload, 96],
              [calldataload, [add, 4, 0]],
              [mload, 64]],
            # Line 2
            [mstore, 352 <z>, [mload, 320 <y>]],
            # Line 1
            stop]]],
      0]]]
['_sym_2', 'JUMP', '_sym_1', 'BLANK', ['PUSH1', 0, 'CALLDATALOAD', 'PUSH1', 28, 'MSTORE', 'PUSH21', 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'PUSH1', 32, 'MSTORE', 'PUSH16', 127, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 'PUSH1', 64, 'MSTORE', 'PUSH32', 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 128, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'PUSH1', 96, 'MSTORE', 'PUSH21', 1, 42, 5, 241, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 253, 171, 244, 28, 0, 'PUSH1', 128, 'MSTORE', 'PUSH32', 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 254, 213, 250, 14, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'PUSH1', 160, 'MSTORE', 'PUSH4', 217, 143, 254, 17, 'PUSH1', 0, 'MLOAD', 'EQ', 'ISZERO', '_sym_3', 'JUMPI', 'PUSH1', 32, 'PUSH1', 4, 'PUSH2', 1, 64, 'CALLDATACOPY', 'CALLVALUE', 'ISZERO', 'ISZERO', 'PC', 'JUMPI', 'PUSH1', 96, 'MLOAD', 'PUSH1', 0, 'PUSH1', 4, 'ADD', 'CALLDATALOAD', 'DUP1', 'PUSH1', 64, 'MLOAD', 'SWAP1', 'SGT', 'PC', 'JUMPI', 'DUP1', 'SWAP2', 'SWAP1', 'SLT', 'PC', 'JUMPI', 'POP', 'PUSH2', 1, 64, 'MLOAD', 'PUSH2', 1, 96, 'MSTORE', 'STOP', '_sym_3', 'JUMPDEST'], '_sym_2', 'JUMPDEST', '_sym_1', '_sym_2', 'SUB', '_sym_1', 'PUSH1', 0, 'CODECOPY', '_sym_1', '_sym_2', 'SUB', 'PUSH1', 0, 'RETURN']
0x6100d556600035601c52740100000000000000000000000000000000000000006020526f7fffffffffffffffffffffffffffffff6040527fffffffffffffffffffffffffffffffff8000000000000000000000000000000060605274012a05f1fffffffffffffffffffffffffdabf41c006080527ffffffffffffffffffffffffed5fa0e000000000000000000000000000000000060a05263d98ffe1160005114156100d057602060046101403734151558576060516000600401358060405190135857809190125857506101405161016052005b5b6100046100d5036100046000396100046100d5036000f3
